File: Lab01/C_Lang/src/Demo15.cpp
//
//  Created by NDHuy at 12/04/2020
//  Copyright 2020 by SEstudio
//  Check increment or decreament digit
//

#include <stdio.h>

int main()
{
    int n; 
    printf("Enter a integer n > 0: ");
    scanf("%d", &n);

    if (n <= 0)
        printf("%d is not greater than Zero\n", n);
    else
    {
        if (n % 10 == 0)
            printf("%d is TRUE with one digit\n", n);
        else
        {
            int t = n % 10;
            int k = (n / 10) % 10;

            bool isIncrement = (t > k) ? true : false;
            int i;
            for(i = n / 10; i > 0; i /= 10)
            {
                k = i % 10;

                if (isIncrement)
                {
                    if (t < k)
                        break;    
                }
                else
                {
                    if (t > k)
                        break;
                }
                
                t = k;
            }

            if (i > 0)
                printf("%d is NOT increment or decrement\n", n);
            else
                if (isIncrement)
                    printf("%d is INCREMENT digit\n", n);
                else
                    printf("%d is DECREMENT digit\n", n);
        }
    }

    return 0; 
}

File: Lab01/Python_Lang/Demo15.py
#
#  Created by NDHuy at 12/04/2020
#  Copyright 2020 by SEstudio
#  Check increment or decreament digit
#

def main():
    n = int(input("Enter a integer number n > 0: "))

    if (n < 0):
        print("%d is not greater than Zero" %(n))
    else:
        t = n % 10
        k = int(n / 10) % 10

        isIncrement = True if (t > k) else False

        i = int(n / 10)
        while(i > 0):
            k = i % 10

            if isIncrement:
                if (t < k):
                    break
            else:
                if (t > k):
                    break

            t = k
            i = int(i / 10)

        if (i > 0):
            print("%d is NOT increment or decrement" %(n))
        else:
            if (isIncrement):
                print("%d is INCREMENT digit" %(n))
            else:
                print("%d is DECREMENT digit" %(n))

if __name__ == "__main__":
    main()